### ISSUE TITLE
Bug: @typescript-eslint/consistent-type-imports - fixStyle is not triggering its error

### ISSUE THREAD

**Issue by silversonicaxel** [typescript-eslint/typescript-eslint#8536]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.3.3&fileType=.tsx&code=JYWwDg9gTgLgBDAnmApnA3nAKslAxKCEAGWACM4BfOAM0JDgCICjSzGg&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6MgeyeUuX0Ra0ipWpQC2xDtHyoMAbXDYciaNCmQANIqVYl2SABNeAQ3jwOAdwAqJRAEEmTDvmP5KXWWHxxEWvfoAzSgAPAGUiJHQoZERiY2hXRCFbUQkpGU1tPUhiaEQAlSjIYSTxSWlULIgAXyyAXUVa6qA&tsconfig=N4XyA&tokens=false

### Repro Code

```TypeScript
import { type TypeFromLib } from "FromLib"
```

This line should be fixed with the relative ESLint configuration, but it doesn't.

The problem lives in both `fixStyle` configuration, `separate-type-imports` and `inline-type-imports`.

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  {
    "rules": {
      "@typescript-eslint/consistent-type-imports": [
        "error", {
          "disallowTypeAnnotations": true,
          "fixStyle": "separate-type-imports",
          "prefer": "type-imports"
        }
      ]
    }
  }
};
```

### Expected Result

```
import type { TypeFromLib } from "FromLib"
```

### Actual Result

```
import { type TypeFromLib } from "FromLib"
```

**Comment by bradzacher** [comment]
As it's name suggests - `fixStyle` only influences the fixer.

If you want to specifically enforce all type specifiers are one way or the other, use the [`import/consistent-type-specifier-style`](https://github.com/import-js/eslint-plugin-import/blob/main/docs%2Frules%2Fconsistent-type-specifier-style.md) rule

**Comment by emer7** [comment]
Hi will it be a good idea to update the documentation (https://typescript-eslint.io/rules/consistent-type-imports/#fixstyle) to state more explicitly that `fixStyle` will not trigger error?

**Comment by bradzacher** [comment]
The docs state this, emphasis mine

> This option defines the expected type modifier **to be added** when an import is detected as used only in the type position

And the option is called `fixStyle`, not something like `specifierStyle`.

What additional clarification would you like to see?

I'm not saying we can't add extra clarification - I'm just looking to understand for future reference so we can make sure new docs are clearer up front.
